test/gtest/conv_igemm_dynamic_xdlops_half.cpp

```diff
@@ -0,0 +1,160 @@
+#include "../conv2d.hpp"
+#include "get_handle.hpp"
+
+using TestCase = std::tuple<std::vector<std::string>, std::string>;
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_ALL)
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_GPU_XNACK_ENABLED)
+
+static bool SkipTest(void)
+{
+    return miopen::IsEnabled(MIOPEN_TEST_GPU_XNACK_ENABLED{}) ||
+           miopen::IsDisabled(MIOPEN_TEST_ALL{});
+}
+
+void GetArgs(const TestCase& param, std::vector<std::string>& tokens)
+{
+    auto env_vars = std::get<0>(param);
+    for(auto& elem : env_vars)
+    {
+        putenv(elem.data());
+    }
+
+    auto cmd = std::get<1>(param);
+
+    std::stringstream ss(cmd);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class Conv2dHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case miopenHalf: params = Conv2dHalf::GetParam(); break;
+    case miopenFloat:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+    case miopenFloat8:
+    case miopenBFloat8:
+        FAIL() << "miopenFloat, miopenInt8, miopenBFloat16, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data type not supported by conv_igemm_dynamic_xdlops_half test";
+
+    default: params = Conv2dHalf::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(),
+                       tokens.end(),
+                       std::back_inserter(ptrs),
+                       [](const std::string& str) { return str.data(); });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx908" || devName == "gfx90a" || miopen::StartsWith(devName, "gfx94"))
+        return true;
+    else
+        return false;
+}
+
+TEST_P(Conv2dHalf, HalfTest)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenHalf);
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+std::vector<TestCase> GetTestCases(const std::string& precision)
+{
+
+    std::vector<std::string> env_fwd = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicFwdXdlops"};
+
+    std::vector<std::string> env_wrw = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicWrwXdlops"};
```

Comment:
This seems not working as expected, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2409#issuecomment-1788093364

Reply:
[resolved]